### PR TITLE
[iOS] Implemented tooltip

### LINKED ISSUE #5473: [iOS] Renderer support for tooltip property

### PR DESCRIPTION
**Pull request by jwoo-msft**
# Related Issue

Fixed #5473 

# Description

Implemented tooltip & changed long press gesture to tap gesture.

# How Verified

Ran unit tests and manually tested

Screen Shot 2021-09-24 at 6 03 32 PM
.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/AdaptiveCards/pull/6409)

### CODE REVIEW

**Review by paulcam206 — APPROVED** [review]

**Comment by paulcam206** [comment]
needs CI pipeline update to be able to consume fluent package

**Review by paulcam206 — CHANGES_REQUESTED** [review]
needs CI in place before we can take this change

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACOBaseActionElement.mm [hunk 1]
```diff
@@ -189,4 +191,38 @@ + (NSNumber *)getKey:(ACRActionType)actionType
...
+        _tooltip = [NSString stringWithCString:_elem->GetTooltip().c_str() encoding:NSUTF8StringEncoding];
+        return _tooltip;
+    }
+
+    return nil;
+}
+
+- (void)setTooltip:(NSString *)tooltip
+{
+    if (tooltip) {
```
Comment (almedina-ms) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACOBaseActionElement.mm:210]: why is the tooltip string emptiness not verified as in line 200?

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRContentStackView.mm [hunk 2]
```diff
@@ -557,20 +557,37 @@ - (CGFloat)getViewWithMaxDimensionAfterExcluding:(UIView *)view dimension:(CGFlo
...
 - (void)configureForSelectAction:(ACOBaseActionElement *)action rootView:(ACRView *)rootView
 {
     if (action != nullptr) {
-        NSObject<ACRSelectActionDelegate> *target = nil;
+        ACRBaseTarget *target = nil;
         if (ACRRenderingStatus::ACROk == buildTarget([rootView getSelectActionsTargetBuilderDirector], action, &target)) {
             [self addTarget:target];
             self.selectActionTarget = target;
             setAccessibilityTrait(self, action);
+            self.accessibilityLabel = configureForAccessibilityLabel(action, [self gatherAccessibilityLabelsFromSubviews]);
```
Comment (almedina-ms) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRContentStackView.mm:573]: why not setting nil in here? is this method overriden by other classes?

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRBaseTarget.h [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+#import "ACOBaseActionElement.h"
+#import "ACRTapGestureRecognizerEventHandler.h"
+#import <UIKit/UIKit.h>
+
+@interface ACRBaseTarget : NSObject <ACRSelectActionDelegate>
+
+- (void)addGestureRecognizer:(UIView *)view tooTipText:(NSString *)toolTipText;
+
+- (void)setTooltip:(UIView *)view tooTipText:(NSString *)toolTipText;
```
Comment (almedina-ms) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRBaseTarget.h:16]: typo: toolTipText

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRBaseTarget.h [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+//  Copyright © 2021 Microsoft. All rights reserved.
+//
+
+#import "ACOBaseActionElement.h"
+#import "ACRTapGestureRecognizerEventHandler.h"
+#import <UIKit/UIKit.h>
+
+@interface ACRBaseTarget : NSObject <ACRSelectActionDelegate>
+
+- (void)addGestureRecognizer:(UIView *)view tooTipText:(NSString *)toolTipText;
```
Comment (almedina-ms) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRBaseTarget.h:14]: >tooTipText [](http://example.com/codeflow?start=44&length=10)

same type as in line 16

**Review by almedina-ms — CHANGES_REQUESTED** [review]
:clock1:

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACOBaseActionElement.mm (hunk 1 above)
Comment (jwoo-msft) [reply source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACOBaseActionElement.mm:210]: It is verified. _tooltip is the property name accessed within an object.

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRBaseTarget.h (hunk 4 above)
Comment (jwoo-msft) [reply source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRBaseTarget.h:14]: nice catch

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRContentStackView.mm (hunk 2 above)
Comment (jwoo-msft) [reply source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRContentStackView.mm:573]: Good question. iOS doesn't support synthesizing accessibility labels from its content views or subviews.
That function will be implemented in our SDK in the future and the synthesized label will be accessed through the method that just returns nil.

**Review by paulcam206 — APPROVED** [review]

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRBaseTarget.h (hunk 4 above)
Comment (almedina-ms) [reply source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRBaseTarget.h:14]: still not fixed here

**Review by almedina-ms — APPROVED** [review]
:shipit:

**Comment by paulcam206** [comment]
build failure has to do with broken symlinks in the PR -- I'm trying to fix now

**Comment by almedina-ms** [comment]
uwp builds are failing due to an expired certificate in one of the uwp projects

https://github.com/microsoft/AdaptiveCards/pull/6470

**Comment by jwoo-msft** [comment]
been working on fixing unit test breakage caused by pod dependency.   didn't realize PR was auto merged. created a new pr.

**Pull request merged**